### ISSUE TITLE
Corrupted renders on iOS devices are not completely fixed

### ISSUE THREAD

**Issue by Aurelain** [mathjax/MathJax#3023]
### Issue Summary

The problem reported in https://github.com/mathjax/MathJax/issues/2435 still occurs in some instances.
As can be seen in the following image, the fix from issue 2435 is applied (`mjx-container{will-change: opacity}`), but the expression is still partly occluded:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/2912123/225709058-2f7a25f7-a1e8-49e3-b803-5d1db1aed9cb.png">

I believe the initial fix (the one from [2021-06-09](https://github.com/mathjax/MathJax-src/pull/722/commits), with `mjx-c::before{will-change: opacity}`) was a bit better, because for my case it fixes 99% of the problem.

### Steps to Reproduce:
1. Host the attached html somewhere
2. View it on an iOS
3. Click the button to toggle scale

There are several ways of fixing this... (I counted 9 methods, see the comments in the html).

### Technical details:

* MathJax Version: 3.2
* Client OS: iOS 14.6
* Device: iPhone 12 (on Browserstack)
* Browser: Safari 14.0

I am using the following html file:
``` html
<!DOCTYPE html>
<html lang='en'>
    <head>
        <meta charset='utf-8' />
        <meta name='viewport' content='width=device-width, initial-scale=1' />
        <title>iOS MathJax Corruption</title>
        <script src='https://polyfill.io/v3/polyfill.min.js?features=es6'></script>
        <script id='MathJax-script' async src='https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js'></script>
        <style>
            #test {
                width: 100px;
                height: 100px;
                transform-origin: 0 0;
                /* Comment the following line to fix the problem. However, why would we accept this fix? */
                transition: transform 1ms;
            }

            .scaled {
                /* Comment the following line to fix the problem. However, why would we accept this fix? */
                transform: scale(4);
            }

            html {
                box-sizing: border-box;
                /* Comment the following line to fix the problem. However, why would we accept this fix? */
                font-family: sans-serif;
            }

            *, *::before, *::after {
                /* Comment the following line to fix the problem. However, why would we accept this fix? */
                /* Note: If we replace the value with `border-box`, the problem is fixed. Unbelievable. */
                box-sizing: inherit;
            }

            /* Uncomment the following line to fix the problem. However, the fix has a lag (happens after 1 second).*/
            /*mjx-container { will-change: unset !important; }*/

            /* Uncomment the following line to fix the problem. However, the top of the "O" is a bit cut.*/
            /*mjx-c::before { will-change: opacity; }*/

            /* Uncomment the following line to fix the problem. However, all symbols start snapping to pixels
            producing an ugly baseline asymmetry in some instances.*/
            /*mjx-c::before { transform: translateZ(0); }*/
        </style>
    </head>
    <body>
        <!-- If we add the symbol "1", the problem is fixed. However, why would we accept this fix? -->
        <!-- If we use inline math, the problem is fixed. However, why would we accept this fix? -->
        <div id='test'>$$\text{O}^2$$</div>
        <button>Click to toggle scale</button>
        <script>
            document.querySelector('button').addEventListener('click', () => {
                document.getElementById('test').classList.toggle('scaled');
            });
        </script>
    </body>
</html>
```

**Comment by dpvc** [comment]
I'm wondering if you would be willing to try out the v4.0.0-alpha.1 release and see if the problem persists there, and if any of your suggestions help?  Version 4 has significant changes to the font handling (new fonts, using the characters in-line instead of in CSS `content`, change in the font ascent and descent values, etc.) any of which might improve the situation.

To do so, use `@4.0.0-alpha.1` in the URL for MathJax rather than `@3`.

Thanks!

**Comment by dpvc** [comment]
There is a choice of fonts for v4, see the [release notes](https://github.com/mathjax/MathJax-src/releases/tag/4.0.0-alpha.1#fonts) for a list, and for how to configure MathJax to select a given font.  Note that the original MathJax TeX fonts are still available (repackaged, but the same glyphs) so you can go back to those if you wish.  They don't have the coverage of the other fonts, but are there for those who don't want to change the look of their sites.  The new mathjax-modern font (based on Latin Modern) is a bit thin for my taste, as well.

**Comment by dpvc** [comment]
PS, thanks for trying it and letting us know that it seems to fix the issue.

**Comment by dpvc** [comment]
PPS, just to be clear, was that with or without the CSS that you listed above?

**Comment by Aurelain** [comment]
Here's a summary of the findings in this thread:
1. Switching to **v4.0.0-alpha.1** fixes the rendering problem, without changing any of the CSS. That means v4 renders properly, with or without any box-sizing/will-change hacks.
2. With v4, `:root mjx-container{will-change:opacity}` doesn't seem needed anymore. This could be a TODO for you (to remove it from the codebase, if the situation in https://github.com/mathjax/MathJax/issues/2435 doesn't warrant it anymore).
3. With v4, using `window.MathJax = {output: {font: 'mathjax-tex'}};` the display is mostly preserved on Chrome Desktop:
![MathJax](https://user-images.githubusercontent.com/2912123/225793649-b00a1c30-5ef5-45a9-af0a-47b74db5f5a4.gif)
4. Using `transform:translateZ(0)` should be avoided at all costs, because it produces pixel snapping/giggle on both v3 and v4:
<details>
  <summary>Without translateZ (good):</summary>

![PixelStatic](https://user-images.githubusercontent.com/2912123/225793850-55ae3b9e-ee5e-4e11-b869-7e58b4b83b0d.gif)
</details>
<details>
  <summary>With translateZ (bad):</summary>

![PixelDance](https://user-images.githubusercontent.com/2912123/225794113-6f3f98ea-e0fa-472d-812c-f1019b6b3ceb.gif)
</details>

In conclusion, possible solutions to the current occlusion problem:
- tinker with the CSS (one of the 9 methods I described in the html comments)
- switch to v4

Thanks for the support!

**Comment by dpvc** [comment]
Thanks for the thorough investigation, and the details.  We appreciate your help in analyzing this situation for both v3 and v4!

**Comment by dpvc** [comment]
I've made a PR to remove the Safari hack.
